### DISCUSSION TITLE
Update actors dynamically without creating a new one in GUI

### DISCUSSION THREAD

**Discussion by zhengliuX** [pyvista/pyvista#1950] [Q&A]
Hi, I'm building a GUI and some interactive actions are needed, like updating the color, or the opacity. Now every time I did this, I created a new mesh, which I think is now efficient. So is there any way that could help me directly change some of the properties of the actor without creating a new mesh?

**Comment by jsaintauret** [comment]
Hello,

You should take a look at the exemple in the following page : https://qtdocs.pyvista.org/usage.html but basically you will need to use the BackgroundPlotter from
pyvista.

Here also [#139](https://github.com/pyvista/pyvista-support/issues/139) you'll find more detail on these methods.

**Reply by zhengliuX** [reply]
Hello, I did use that in the GUI. What I want to do is to update the actor without creating a new mesh, like changing the opacity.

**Comment by adam-grant-hendry** [comment]
@akaszynski @banesullivan Can you assist? I have the same question.

**Reply by jsaintauret** [reply]
I think one way you could do it, is to get the actor from your renderer.

Something like that might work:

pl = pyvista.Plotter()
renderer = pl.renderrer 
actor = renderer.actors

In that case the actor is a dictionary of the actors assign to your renderer

prop = actors[id].GetProperty()
prop.SetOpacity(opacity)

make sure your plotter is an instance of BackgroundPlotter

**Reply by banesullivan** [reply]
Yep, you'll need to capture the vtkActor and apply your operations to that: the actor can be updated to change colors, turn on/off visibility, etc.

While that dictionary from `renderer.actors` is useful, you may find it more intuitive to grab the actor as it is returned by `add_mesh()`
